lib/awful/rules.lua

```diff
@@ -324,8 +325,16 @@ function rules.high_priority_properties.new_tag(c, value)
         -- Create a tag named after "value"
         t = atag.add(value, {screen=c.screen, volatile=true})
     elseif ty == "table" then
-        -- Assume a table of tags properties
-        t = atag.add(value.name or c.class or "N/A", value)
+        -- Assume a table of tags properties. Set the right screen, but
+        -- avoid editing the original table
+        local values = value.screen and value or util.table.clone(value)
+        values.screen = values.screen or c.screen
+
+        t = atag.add(value.name or c.class or "N/A", values)
+
+        -- In case the tag has been forced to another screen, move the client
+        c.screen = t.screen
+        props.screen = t.screen -- In case another rule query it
```

Comment:
>  Following executions will see this modification.

That's exactly the idea. The `screen` property has higher priority than this, so it has already been executed. If a further rule query the `props` array to check the screen, it will get the wrong one.

This table is generated by the code, it doesn't add garbage anywhere else. So better fix it to limit the side effects of the other rules.